api/util/GenerateTokenScript.js

```diff
@@ -0,0 +1,127 @@
+const fs = require('fs');
+const readline = require('readline');
+const { google } = require('googleapis');
+
+const redColor = '\x1b[31m';
+const greenColor = '\x1b[32m';
+const blueColor = '\x1b[34m';
+const yellowColor = '\x1b[33m';
+const defaultColor = '\x1b[0m';
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+/**
+ * Prompts user for input and resolves the input or false
+ * @param {String} prompt question to get the info we need
+ */
+const inputPrompt = (prompt) => {
+  return new Promise((resolve, reject) => {
+    rl.question(prompt, (ans) => {
+      if(ans !== '') {
+        resolve(ans);
+      } else {
+        resolve(false);
+      }
+    });
+  });
+};
+
+/**
+ * Prompts user for client ID & secret and writes them to config
+ * file containing auth credentials, then generates a token.json file for GCP
+ * @param {String} configPath the directory path of the config.json file
+ */
+const setAuthCredentials = async (configPath) => {
+  console.debug(blueColor +
+    'Welcome to SCE\'s Google Cloud API Token Generator!\n'
+  );
+  console.debug(
+    yellowColor +
+    'Don\'t know what to do or how this works?\n' +
+    /* eslint-disable-next-line */
+    'Please visit: https://github.com/SCE-Development/Core-v4/wiki/Getting-Started#setting-up-mailer\n' + 
+    defaultColor
+  );
+
+  const rawConfig = fs.readFileSync(configPath);
+
+  if(rawConfig) {
+    const clientID = await inputPrompt('Please enter the Client ID: ');
+    const clientSecret = await inputPrompt('Please enter the Client Secret: ');
+
+    let config = JSON.parse(rawConfig);
+    if(clientID) {
+      config.googleApiKeys.CLIENT_ID = clientID;
+    }
+    if(clientSecret) {
+      config.googleApiKeys.CLIENT_SECRET = clientSecret;
+    }
+
+    fs.writeFile(configPath, JSON.stringify(config), (err) => {
+      if(err) return console.debug('ERR', err);
+      console.debug(greenColor +
+        'Successfully written to:', configPath + defaultColor);
+
+      generateToken(
+        config,
+        ['https://mail.google.com/',
+          'https://www.googleapis.com/auth/calendar'],
+        __dirname + '/../config/token.json',
+      );
+    });
+
+  } else {
+    console.debug(redColor +
+      'config.json file does not exist at path:', configPath
+    );
+  }
+};
+
+/**
+ * Generates an auth token to make API calls, assuming configData exists
+ * @param {Object} configData data from config file containing auth credentials
+ * @param {String | String[]} apiScopes which API(s) to interact with
+ * @param {String} tokenPath path that the auth token file will be written to
+ */
+const generateToken = async (configData, apiScopes, tokenPath) => {
```

Comment:
is there a way you can incorporate this function into [`api/google_api/util/SceGoogleApiHandler.js`](https://github.com/SCE-Development/Core-v4/blob/dev/api/google_api/util/SceGoogleApiHandler.js)? it would be good to isolate the logic into that function.